projects/01-rubygems-cli/solution/lib/commands/results/search_command_result.rb

```diff
@@ -0,0 +1,19 @@
+require "./lib/commands/results/command_result"
+
+class SearchCommandResult < CommandResult
+  attr_accessor :gems
+
+  def initialize(gems, exit_code = 0)
+    super(exit_code)
+    @gems = gems
+  end
+
+  def output
+    result = ""
+    gems.each_with_index do |gem, i|
+      result += "#{i + 1}. #{gem.name}\n"
```

Comment:
You can do this instead:
```ruby
[1,2,3].each.with_index(1) { |obj, index| puts index }
=> 1
2
3
```

Try that in combination with `#map` to get rid of that variable.

Reply:
This is why I found this PR :)

Been Rubying since 2006, just saw this for the first time. Unless I learned it a long time ago, then forgot?